packages/model/src/utils/DeepReadonly.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Make all properties in T readonly
+ */
+export type DeepReadonly<T> =
+    T extends (infer R)[] ? DeepReadonlyArray<R> :
+    // eslint-disable-next-line @typescript-eslint/ban-types
```

Comment:
You can remove this if you use `(...args: unknown[]) => unknown` instead of Function